Sources/ReactiveArray.swift

```diff
@@ -0,0 +1,137 @@
+import Foundation
+import ReactiveSwift
+import enum Result.NoError
+
+public enum Change<T> {
+    case insert(element: T, at: Int)
+    case remove(element: T, at: Int)
+}
+
+public final class ReactiveArray<Element> {
+
+    fileprivate var elements: ContiguousArray<Element>
+
+    fileprivate let innerObserver: Observer<[Change<Element>], NoError>
+
+    public let signal: Signal<[Change<Element>], NoError>
+
+    public var capacity: Int {
+        return elements.capacity
+    }
+
+    public init(_ elements: [Element]) {
+        self.elements = ContiguousArray(elements)
+
+        (signal, innerObserver) = Signal<[Change<Element>], NoError>.pipe()
+    }
+
+    deinit {
+        innerObserver.sendCompleted()
+    }
+
+}
+
+extension ReactiveArray: ExpressibleByArrayLiteral {
+
+    public convenience init(arrayLiteral elements: Element...) {
+        self.init(elements)
+    }
+}
+
+extension ReactiveArray: MutableCollection {
+
+    public typealias SubSequence = ArraySlice<Element>
+    public typealias Index = Int
+
+    public var startIndex: Index {
+        return elements.startIndex
+    }
+
+    public var endIndex: Index {
+        return elements.endIndex
+    }
+
+    public subscript(position: Int) -> Element {
+        get {
+            return elements[position]
+        }
+        set {
+            replaceSubrange(position..<position, with: CollectionOfOne(newValue))
+        }
+    }
+
+    public subscript(bounds: Range<Int>) -> SubSequence {
+        get {
+            return elements[bounds]
+        }
+        set {
+            replaceSubrange(bounds, with: newValue)
+        }
+    }
+}
+
+extension ReactiveArray: RandomAccessCollection {
+
+    public typealias Indices = CountableRange<Int>
+}
+
+extension ReactiveArray: RangeReplaceableCollection {
+
+    public convenience init() {
+        self.init([])
+    }
+
+    public convenience init(repeating repeatedValue: Element, count: Int) {
```

Comment:
I thought about this a little more and I think it's more correct provide our own implementation because the default implementation initializes an instance and then appends each `repeatedValue` which will trigger changes in our signal, which despite not being really harmful, signal won't be available for the outside world before initialization completes, it's inconsistent with our designated initializer which won't trigger any update.